### DISCUSSION TITLE
Custom icon theme support?

### DISCUSSION THREAD

**Discussion by Clasurol** [FelixdelasPozas/TrayWeather#124] [Ideas]
First off: thank you for this wonderful program. It's fantastic and I can't thank you enough. 

Now, on to my idea: is it at all possible for custom theme support for the tray icons? I use classic theme and a pretty heavily modded system made to resemble Windows 2000, and tbh, the included themes kinda don't gel with my setup. I know that sounds silly, but I'd really like to change them to something that goes with the era I'm aiming at.

![Screenshot_2](https://user-images.githubusercontent.com/46195405/180623674-8b6c4b50-4d6e-434e-b246-5de2975ffec7.png)

I know my reason for suggesting the idea is a bit silly, but I think such a feature would benefit others too.

 Thank you for reading and have a good day.

**Comment by FelixdelasPozas** [comment]
Hi @Clasurol .

I stopped looking for icon themes after adding the first batch, but if you have a theme that covers [all icon variants](https://openweathermap.org/weather-conditions#Weather-Condition-Codes-2) and it's free to distribute  (with credit to the author, of course) I'll be glad to add it. 
Your case is a little special, as icons for win 95 will require a "old look" format. In TrayWeather icons a scalable SVG format so it can be scaled in a smooth way for whatever resolution is needed, your look will require a "pixelated" look or it wouldn't look right. But its not much effort to do. Maybe I can also look into adding a way to the user to specify custom icon themes. 
But the "error" and "waiting for data" icons will be SVG for whatever icon theme that is selected because those icons are not part of any theme, just standard application icons. 

Regards, 
Félix.

**Reply by travy-patty** [reply]
I'm wondering, would the icons still be legible if they were scaled up to something like 256x256 with nearest neighbor?

**Reply by Clasurol** [reply]
I've deleted my previous posts. I've spent quite a bit of time looking for old skool style SVG icon packs that could look good, but I'm afraid I've turned up empty. I found a few good packs on here, but they aren't in SVG format, and trying to convert them was a no-go (errors arouse due to the shift in formats).

Thank you very much for your time, and thank you even more for this brilliant program.

**Comment by FelixdelasPozas** [comment]
Hi @Clasurol . I've been using SVG because it scales very well. But the application could use BMP, PNG, ICO or other formats for the icons (usually formats that can have a transparent background color, JPG does not have that for example). OpenWeatherMap [provides icons in PNG format](https://openweathermap.org/weather-conditions) in day and night versions with transparent background.
If you want it to look "old school" you could scale the icon down to, for example 32x32 resolution and scale it back up to 100x100 without using a resampling algorithm (using nearest neighbor as @travy-patty suggested).
![02d@2x-comparison](https://user-images.githubusercontent.com/12167134/190711562-dadc917a-4ffb-4c7d-8575-f6558f024a6b.png)
But letting the user specify custom icons will require a new dialog so it's not some quick fix. And some options for the new formats (like not use smooth scaling if the user doesn't want to). I'll add it to the to-do and try to add it when I have time. 
That way you could use icons not only in SVG format. 

Regards, 
Félix.

**Reply by Clasurol** [reply]
Thank you very much. Take all the time you need, I'm just grateful you even listened to my idea/request in the first place.
